### PR TITLE
Handle routing exceptions during news and event URL generation

### PR DESCRIPTION
**Pull request by fritzmg**
This is a first draft on how #6797 could be fixed - in this case for news as an example. @aschempp is this what you had in mind?

### CODE REVIEW

File: news-bundle/src/Routing/NewsResolver.php [hunk 1]
```diff
@@ -40,19 +40,31 @@ public function resolve(object $content): ContentUrlResult|null
...
             // Link to an article
             case 'article':
                 $articleAdapter = $this->framework->getAdapter(ArticleModel::class);
 
-                return ContentUrlResult::redirect($articleAdapter->findPublishedById($content->articleId));
+                if (!$article = $articleAdapter->findPublishedById($content->articleId)) {
+                    return null;
+                }
+
+                return ContentUrlResult::redirect($$article);
```
Comment (leofeyer) [inline news-bundle/src/Routing/NewsResolver.php]:
```suggestion
                return ContentUrlResult::redirect($article);
```

**Comment by fritzmg** [comment]
As discussed today we should instead catch the routing's `ExceptionsInterface` and also do not fill the `more` template variable with the HTML content of a link in case the URL could not be generated.

**Comment by fritzmg** [comment]
FAQ and newsletters already handled routing exceptions during URL generation (and fall back to the current request URL), so I haven't touched those.

File: news-bundle/contao/modules/ModuleNews.php [hunk 2]
```diff
@@ -80,19 +81,26 @@ protected function parseArticle($objArticle, $blnAddArchive=false, $strClass='',
...
+		$objTemplate->linkHeadline = $objArticle->headline;
 		$objTemplate->archive = $objArticle->getRelated('pid');
 		$objTemplate->count = $intCount; // see #5708
 		$objTemplate->text = '';
 		$objTemplate->hasTeaser = false;
 		$objTemplate->hasReader = true;
 
+		if (null !== $url)
+		{
+			$objTemplate->linkHeadline = $this->generateLink($objArticle->headline, $objArticle, $blnAddArchive);
```
Comment (aschempp) [inline news-bundle/contao/modules/ModuleNews.php:100]: shouldn't the `generateLink` method handle the case of not having a link?

Comment (fritzmg) [reply news-bundle/contao/modules/ModuleNews.php:100]: I opted for not calling the method, if the generation already failed.

Comment (fritzmg) [reply news-bundle/contao/modules/ModuleNews.php:100]: Also I did not want to change the return type of the protected function.

**Review by aschempp — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**